src/fundamental/macro-fundamental.h

```diff
@@ -376,3 +376,15 @@ static inline size_t ALIGN_TO(size_t l, size_t ali) {
         (v) = UPDATE_FLAG(v, flag, b)
 #define FLAGS_SET(v, flags) \
         ((~(v) & (flags)) == 0)
```

Comment:
why would the macro only work for basic types? i don't get this? i mean, the macro would just expand to the right stuff either way? can you explain?

Reply:
and i guess structs are actually fine, but anonymous ones are not?

Reply:
Yeah, anonymous structure is a problem. We *could* make it non-anonymous, but I think we don't want to do that. I'll adjust the commit message.